src/security/KeyData.cc

```diff
@@ -151,14 +146,22 @@ Security::KeyData::loadX509PrivateKeyFromFile()
 
 #if USE_OPENSSL
     const char *keyFilename = privateKeyFile.c_str();
+    Ssl::BIO_Pointer bio;
+    if (!Ssl::OpenCertsFileForReading(bio, keyFilename)) {
+        const auto x = ERR_get_error();
+        debugs(83, DBG_IMPORTANT, "ERROR: unable to load key file '" << keyFilename << "': " << ErrorString(x));
+        return false;
+    }
     // XXX: Ssl::AskPasswordCb needs SSL_CTX_set_default_passwd_cb_userdata()
     // so this may not fully work iff Config.Program.ssl_password is set.
     pem_password_cb *cb = ::Config.Program.ssl_password ? &Ssl::AskPasswordCb : nullptr;
-    Ssl::ReadPrivateKeyFromFile(keyFilename, pkey, cb);
-
-    if (pkey && !X509_check_private_key(cert.get(), pkey.get())) {
-        debugs(83, DBG_IMPORTANT, "WARNING: '" << privateKeyFile << "' X509_check_private_key() failed");
-        pkey.reset();
+    while(!pkey && Ssl::ReadPrivateKey(bio, pkey, cb)) {
```

Comment:
If feels like we are fixing a bug or adding a feature here: We are replacing a single check with a loop. Should we disclose this change in the PR description?

Reply:
I removed this feature because of the next request-for-change

Some notes about:
Imagine the case you have a file with a number of pkeys for various certificates. Why not the administrator use one file for all?
However in my implementation, if the stored pkeys ware using different passphrase then the ReadPrivateKey would return false and the read would abort probably before  the correct pkey read.

A better implementation of ReadPrivateKey should return the type of error and if the error is decoding error we should continue to the next read, or we can add a flag where the decoding errors will be ignored.

The system admin  was not able to overcome the above passprhase problem using the correct sslpassword_program, because this program gets only the key filename as  argument when runs.
Maybe we can fix it and pass an index as argument, or maybe the related certificate subject.

Reply:
I agree that supporting private key bundles may be useful.

If Squid did not support private key bundles then we should not add support for private key bundles in this PR, especially when adding such support is far from trivial. Let's wait for a real use case to guide us. Thank you for removing this new feature. I am resolving this change request.